torchmultimodal/modules/layers/transformer_decoder.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torch import nn
+from torch.utils.checkpoint import checkpoint
+
+from torchmultimodal.modules.layers.attention import FullAttention, MultiHeadAttention
+from torchmultimodal.modules.layers.mlp import MLP
+
+
+class SiLU(nn.Module):
+    r"""Sigmoind Linear Unit
+
+    .. math:: \text{SiLU}(x) = x * \sigma(1.702 * x)
+
+    where :math:`\sigma(x)` is the cumulative distribution function for Logistic Distribution.
+
+    Approximation of the exact GeLU implemented by pytorch for greater forward speed.
+    Reference: `"Gaussian error linear units"<https://arxiv.org/pdf/1606.08415.pdf>`_.
+    """
+
+    def forward(self, x):
+        return torch.sigmoid(1.702 * x) * x
+
+
+class RightShift(nn.Module):
+    def __init__(self, embedding_dim):
+        super().__init__()
+        self.embedding_dim = embedding_dim
+        self.sos = nn.Parameter(
```

Comment:
sos = start of sentence? Is this implicitly for text only then? Also any particular reason for the choice of initialization?